### ISSUE TITLE
many commands doesn't work on my device

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#318]
```
I tested the command "ring" can work, but "ring 0" or "ring 100" doesn't work, 
it returns "unknown command" to me.
Also I see many other commands don't work too, like "sc", "flash", "camera" etc.

```

Original issue reported on code.google.com by `pan...@gmail.com` on 25 Feb 2013 at 1:23

**Comment by GoogleCodeExporter** [comment]
```
Forgot to say I am using Nexus 7 which has android 4.2.2 installed
```

Original comment by `pan...@gmail.com` on 25 Feb 2013 at 1:26

**Comment by GoogleCodeExporter** [comment]
```
Can you provide logs during the application startup? If several commands are 
not working, it's probably due to the startup sequence.

BTW when you write "ring 0" you mean "ring:0", right? Else it's your issue.
```

Original comment by `Florent....@gmail.com` on 25 Feb 2013 at 2:02

**Comment by GoogleCodeExporter** [comment]
```
Yes, its my problem for "ring 0" and "ring 100", they works with ":", but I 
think for other commands, it's not my issue, here is the log:

GTalkSMS Preferences
notifiedAddress: xxx
useGoogleMapUrl: true
locale: default
smsMagicWord: GTalkSMS
manuallySpecifyServerSettings: false
publicIntentsEnabled: false
formatResponses: false
batteryNotificationInterval: 10
useOpenStreetMapUrl: false
serviceName: gmail.com
startAtBoot: false
forceMucServer: true
showStatusIcon: false
markSmsReadOnReply: false
useCompression: false
notifyBatteryInStatus: true
notificationIncomingSmsType: same
notifyInMuc: false
login: @jabber.org
startOnPowerConnected: false
smsReplySeparate: false
startOnWifiConnected: true
stopOnPowerDisconnected: false
notifySmsDelivered: false
stopOnWifiDisconnected: false
debugLog: true
smsNumber: 5
connectOnMainscreenShow: true
notifySmsSent: true
publicIntentTokenRequired: true
callLogsNumber: 10
roomPassword: gtalksms
serverHost: talk.google.com
showSentSms: false
serverPort: 5222
mucServer: conference.jwchat.org
notifyApplicationConnection: false
notifyBattery: false
notifyIncomingCalls: true
stopOnPowerDelay: 1
publicIntentToken: secret

GTalkSMS Version: 4.3
Android API: 17
Kernel info: 3.1.10-g05b777c
android-build@vpbs1 (GCC)
) #1 SMP PREEMPT Thu Nov 29 10:35:37 PST 2012

--------- beginning of /dev/log/main
02-25 21:19:13.551 E/gtalksms( 2366): executeCommand() Exception
02-25 21:19:13.551 E/gtalksms( 2366): java.lang.NullPointerException
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.cmd.CameraCmd.setLight(CameraCmd.java:227)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.cmd.CameraCmd.execute(CameraCmd.java:105)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.cmd.CommandHandlerBase.execute(CommandHandlerBase.java:2
55)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.cmd.CommandHandlerBase.execute(CommandHandlerBase.java:2
24)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.MainService.executeCommand(MainService.java:579)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.MainService.handleCommandFromXMPP(MainService.java:680)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.MainService.onHandleIntent(MainService.java:249)
02-25 21:19:13.551 E/gtalksms( 2366): at 
com.googlecode.gtalksms.MainService$ServiceHandler.handleMessage(MainService.jav
a:143)
02-25 21:19:13.551 E/gtalksms( 2366): at 
android.os.Handler.dispatchMessage(Handler.java:99)
02-25 21:19:13.551 E/gtalksms( 2366): at android.os.Looper.loop(Looper.java:137)
02-25 21:19:13.551 E/gtalksms( 2366): at 
android.os.HandlerThread.run(HandlerThread.java:60)
02-25 21:23:49.541 I/gtalksms( 2366): MyBackupAgent onBackup() begin
02-25 21:23:49.541 I/gtalksms( 2366): MyBackupAgent onBackup() new data found - 
starting backup
02-25 21:26:10.941 D/gtalksms( 2366): NetworkConnectivityReceiver: 
com.googlecode.gtalksms.action.NETWORK_STATUS_CHANGED name=WIFI changed=false 
connected=true connectedOrConnecting=true
02-25 21:26:19.091 D/gtalksms( 2366): NetworkConnectivityReceiver: 
com.googlecode.gtalksms.action.NETWORK_STATUS_CHANGED name=WIFI changed=false 
connected=true connectedOrConnecting=true
02-25 22:04:42.831 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/AUTHORS
02-25 22:04:48.641 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/AUTHORS in 5 sec
02-25 22:04:48.641 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/Donors
02-25 22:04:48.861 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/Donors in 0 sec
02-25 22:04:48.861 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/Changelog
02-25 22:04:49.101 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/Changelog in 0 sec
02-25 22:05:55.171 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/AUTHORS
02-25 22:05:55.571 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/AUTHORS in 0 sec
02-25 22:05:55.661 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/Donors
02-25 22:05:55.891 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/Donors in 0 sec
02-25 22:05:55.891 D/gtalksms( 2366): Downloading URL: 
http://gtalksms.googlecode.com/git/Changelog
02-25 22:05:56.151 D/gtalksms( 2366): Downloaded 
http://gtalksms.googlecode.com/git/Changelog in 0 sec
02-25 22:10:17.861 D/gtalksms( 2366): MainActivity: onSart()
02-25 22:10:17.881 D/gtalksms( 2366): MainActivity: MainService connected
02-25 22:10:24.971 D/gtalksms( 2366): MainActivity: onStop()
02-25 22:10:56.221 D/gtalksms( 2366): MainActivity: onSart()
02-25 22:10:56.241 D/gtalksms( 2366): MainActivity: MainService connected
02-25 22:11:21.321 D/gtalksms( 2366): MainActivity: onStop()
```

Original comment by `pan...@gmail.com` on 25 Feb 2013 at 2:15

**Comment by GoogleCodeExporter** [comment]
```
Concerning commands depending on the hardware likes flash, photo, video it 
might failed due to the manufacturer integration. What Android device do you 
have?

The sc command is no more supported and have been removed few version ago due 
to a lot of incompatibilities.
```

Original comment by `Florent....@gmail.com` on 10 Nov 2013 at 9:33
- Changed state: **UserFeedBack**

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 15 Aug 2014 at 7:28
- Changed state: **FixedInBeta**

**Comment by GoogleCodeExporter** [comment]
```
Released on Play Market as 5.0
```

Original comment by `Florent....@gmail.com` on 25 Oct 2014 at 6:23
- Changed state: **Deployed**
